processor/src/main/java/org/mapstruct/ap/internal/model/source/MappingMethodOptions.java

```diff
@@ -177,13 +177,54 @@ public void applyInheritedOptions(SourceMethod templateMethod, boolean isInverse
             }
 
             // now add all (does not override duplicates and leaves original mappings)
-            mappings.addAll( newMappings );
+            addAllNonRedefined( newMappings );
 
             // filter new mappings
             filterNestedTargetIgnores( mappings );
         }
     }
 
+    private void addAllNonRedefined(Set<MappingOptions> inheritedMappings) {
+        Set<String> redefinedSources = mappings.stream()
```

Comment:
Streams are all OK, but we are going twice through the same mappings.

Why not defined the sets, then use a for-loop to iterate the mappings and add to the sets?

Reply:
will correct.. I'm all into this streams nowadays.. 😄

Reply:
I love streams as well. However, good old imperative style is better sometimes. Lately I am all into performance 😀